packages/docs/src/pages/guides/typescript.mdx

```diff
@@ -10,56 +10,54 @@ While most APIs in Theme UI should _just work_ in TypeScript,
 there are a few advanced use cases which will differ slightly.
 This guide is intended to cover those use cases.
 
-## Exact theme type
+## Extending the Theme
 
-The [`Theme` type](/theme-spec) represents all possible themes.
+If you need additional properties in the theme object,
+you can add them with [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).
 
-It might be what you need when you're writing a library of reusable components
-or an app where the theme object is provided by the user and kept in the database.
-However, it's not the most convenient way to think about the theme,
-when building a blog or a landing page.
+```tsx
+import { makeTheme, UserTheme } from '@theme-ui/css';
 
-To know the exact type of your particular theme on type level,
-you can use an identity "constructor" function to narrow the type.
+interface MyTheme {
```

Comment:
Could we use `makeTheme` instead of it? Declaring the type manually could get pretty boring, and we'll export that makeTheme function (which you can still see described on the left side of the diff).

```tsx
import { makeTheme } from 'theme-ui/exact-theme';

type MyTheme = typeof theme
```
```

Reply:
Up to you. I dislike using `typeof` this way because it's kind of "circular". If there's a bug in the theme, it will "bubble" to the type. There might also be problems with type widening... I can't think of a specific example but `typeof` could infer `'a' | 'b' | 'c'` as `string` when you _really_ want `'a' | 'b' | 'c'`.

> Could we use makeTheme instead of it?

Maybe "in addition to", instead of "instead of"? 😄

Reply:
> I can't think of a specific example but typeof could infer 'a' | 'b' | 'c' as string when you really want 'a' | 'b' | 'c'.

`makeTheme` narrows the type, so we'll get all properties instead of `string`.

> because it's kind of "circular"

The "graph" in my head looks like this:

```
( BaseTheme - type matching any theme, useful for libraries )
                                               |
                                              V
( user's theme object has to be assignable to BaseTheme, or we wouldn't work )
                                               |
                                              V
( UserTheme - the exact type of user's theme object, useful for websites, apps, blogs - they have just one theme so they don't have to "tread lightly" and use optional chaining everywhere)
                                               |
                                              V
( Theme = Assign<BaseTheme, UserTheme> )
```

Reply:
Let me know if this is what you had in mind: https://github.com/hasparus/theme-ui/pull/4/commits/0452bdac4a5e415865662b3ecf229b88a37d90b9#diff-8ec9599b94c082ff1719d0791a7d7c2ee03ce4da1f699b626352f3595ed9f6b6R32